src/shogun/machine/GLM.cpp

```diff
@@ -23,33 +23,19 @@ using namespace shogun;
 
 GLM::GLM(GLM_DISTRIBUTION distr, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta): IterativeMachine<LinearMachine>()
 {
-	init();
 	distribution=distr;
 	m_alpha=alpha;
 	m_lambda=lambda;
-	m_learning_rate=learning_rate;
 	m_max_iterations=max_iterations;
 	m_tolerance=tolerance;
 	m_eta=eta;
-}
-
-void GLM::init()
-{
-	bias = 0;
-
-	distribution= POISSON;
-	m_alpha=0.5;
-	m_lambda=0.1;
-	m_learning_rate=2e-1;
-	m_max_iterations=1000;
-	m_tolerance=1e-6;
-	m_eta=2.0;
-}
-
 
-GLM::~GLM()
-{
+	m_learning_rate->set_const_learning_rate(2e-1);
+	m_learning_rate->set_const_learning_rate(learning_rate);
```

Comment:
?

Reply:
Sorry, that's an error. Only the second line should be there.